### PR TITLE
Fixes #21843 When asadmin get-client-stubs doesn't find application, …

### LINKED ISSUE #21843: When asadmin get-client-stubs doesn't find application, then command should fail.
When asadmin get-client-stubs subcommand doesn't find application, then command should fail.
But actually, it succeeds.

## Environment Details
* GlassFish Version: 4.1.2, 5.0-b08
* JDK version: 1.8.0_65
* OS: Windows 7

----------

## Problem Description
[//]: # "Describe the bug in detail highlighting current behavior vs expected behavior"
[//]: # "State if the problem is easily reproducible or happens intermittently"
[//]: # "Include stack traces or command outputs"
asadmin get-client-stubs subcommand searches application by name which is provided by --appname parameter.
When executing get-client-stubs with --appname <not-deployed-app-name>, searching application fails and client-stubs doesn't be downloaded.
However, get-client-stubs subcommand returns successful (return code is 0).

```
> asadmin get-client-stubs --appname foobar C:\
Application foobar was not found
Command get-client-stubs executed successfully.

> echo %ERRORLEVEL%
0
```

In this case, I believe that get-client-stubs subcommand should return failure (return code is NOT 0).

## Steps to reproduce
1. asadmin start-domain
1. asadmin get-client-stubs --appname foo C:\ (/tmp for Linux)

## Fix
If application is not found in GetClientStubsCommand, it should report failure.

```diff
diff --git "a/C:\\Users\\uno\\AppData\\Local\\Temp\\GetDB3B.tmp\\GetClientStubsCommand-HEAD-left.java" "b/D:\\repo\\external\\glassfish\\nucleus\\deployment\\admin\\src\\main\\java\\org\\glassfish\\deployment\\admin\\GetClientStubsCommand.java"
index 4d2fb57..662cbd5 100644
--- "a/C:\\Users\\uno\\AppData\\Local\\Temp\\GetDB3B.tmp\\GetClientStubsCommand-HEAD-left.java"
+++ "b/D:\\repo\\external\\glassfish\\nucleus\\deployment\\admin\\src\\main\\java\\org\\glassfish\\deployment\\admin\\GetClientStubsCommand.java"
@@ -105,7 +105,7 @@ public class GetClientStubsCommand implements AdminCommand, AdminCommandSecurity
                 return true;
             }
         }
-        context.getActionReport().setMessage(localStrings.getLocalString(
+        context.getActionReport().failure(context.getLogger(), localStrings.getLocalString(
             getClass(),
             "get-client-stubs.noSuchApp",
             "Application {0} was not found",
```

Applying fix, get-client-stubs subcommand returns failure when application is not found.
```
> asadmin get-client-stubs --appname foobar C:\
remote failure: Application foobar was not found
Command get-client-stubs failed.
　
> echo %ERRORLEVEL%
1
```

### PR DESCRIPTION
**Pull request by mukumawa**
…then command should fail

### CODE REVIEW

**Comment by mukumawa** [comment]
@glassfishrobot Run CI tests please

**Comment by mukumawa** [comment]
@glassfishrobot Run CI tests please

**Review by yaminikb — CHANGES_REQUESTED** [review]
Can you please add a test (under deployment dev tests) for this?

**Comment by mukumawa** [comment]
@glassfishrobot Run CI tests please

**Comment by mukumawa** [comment]
@glassfishrobot Run CI tests please

**Comment by yaminikb** [comment]
@glassfishrobot run ci tests please

**Comment by yaminikb** [comment]
@glassfishrobot run ci tests please

**Comment by mukumawa** [comment]
@glassfishrobot run ci tests please

**Comment by arindam-bandyopadhyay** [comment]
@glassfishrobot run ci tests please

**Comment by mukumawa** [comment]
@glassfishrobot run ci tests please

**Comment by mukumawa** [comment]
@glassfishrobot run ci tests please

File: appserver/tests/appserv-tests/devtests/admin/cli/build.xml [hunk 1]
```diff
@@ -150,7 +150,7 @@
      <target name="cli-group-1" depends="setup, monitoring, zombie, ports, instance,teardown"/>
      <target name="cli-group-2" depends="setup,cluster,configs, sync, domain,teardown"/>
      <target name="cli-group-3" depends="setup,backup, tokens, validation, node,teardown"/>
-     <target name="cli-group-4" depends="setup,logging-command,load-balancer, getset, misc-commands, teardown"/>
+     <target name="cli-group-4" depends="setup, clientstubs, logging-command,load-balancer, getset, misc-commands, teardown"/>
```
Comment (yaminikb) [inline appserver/tests/appserv-tests/devtests/admin/cli/build.xml]: It can also be added to "cli" target which runs all tests.

**Review by yaminikb — CHANGES_REQUESTED** [review]

**Comment by mukumawa** [comment]
@glassfishrobot run ci tests please

File: appserver/tests/appserv-tests/devtests/admin/cli/build.xml (hunk 1 above)
Comment (mukumawa) [inline appserver/tests/appserv-tests/devtests/admin/cli/build.xml]: done

**Review by yaminikb — APPROVED** [review]

**Pull request merged**